### DISCUSSION TITLE
How is the status of Transport maintained?

### DISCUSSION THREAD

**Discussion by Ma233** [libp2p/rust-libp2p#3879] [Q&A] [answered]
Hello, I am a developer of https://github.com/RingsNetwork/rings-node. (It‘s still in the bootstrap phase. It may be a little messy.)

We are trying to build a p2p network using WebRTC. We want to use the excellent rust-libp2p to help with our project development.

While reading the code in `transports/webrtc`, I found that the newly created RTCPeerConnection object does not declare `on_peer_connection_state_change`.

In our project, we use this event function to notify an external module of remote connection closing. Does rust-libp2p use other mechanisms to detect transport disconnection? What happens if a peer closes the connection to a WebRTC transport?

**Comment by thomaseizinger** [comment, marked as answer]
Our WebRTC supports is primarily in alpha state because it has not yet been battle-tested. It may very well be that we don't react closed connections correctly.

Personally, I am not very happy with how `webrtc-rs` is structured and I'd like to move to `str0m`: https://github.com/libp2p/rust-libp2p/issues/3659

If you have WebRTC expertise, any help on that front would be much appreciated.

**Comment by thomaseizinger** [comment]
Also see https://github.com/libp2p/rust-libp2p/discussions/3875 for a similar discussion.
